backend/__tests__/cronJobs.test.ts

```diff
@@ -0,0 +1,122 @@
+            pointOfContact: "contact@openIMIS.com",
+            status: 0,
+            expirationDate: new Date("2022-11-24T00:00:00.000Z"),
+            compliance: [],
+            link: "123456"
+        });
+
+        clock = sinon.useFakeTimers(currentDate.getTime());
+        appConfig.cron.removeExpiredDraftsSchedule = '* * * * * *';
+        startCronJobs();
+    });
+
+    after(async () => {
+        if (clock) {
+            clock.restore();
+        }
+        appConfig.cron.removeExpiredDraftsSchedule = originalCronSchedule;
+        await mongoose.connection.dropDatabase();
+        await mongoose.disconnect();
+        await mongod.stop();
+    });
+
+    it('should execute the cron job and call removeSensitiveDataFromExpiredDrafts multiple times', async () => {
+        const stub = sinon.stub(removeExpiredDrafts, 'removeSensitiveDataFromExpiredDrafts');
+        
+        for (let i = 0; i < 5; i++) {
+            await clock.tickAsync(1000);
+        }
+
+        expect(stub.callCount).to.be.greaterThan(1);
+        stub.restore();
+    });
+
+    it('should actually remove sensitive data from expired drafts', async () => {
+        await removeExpiredDrafts.removeSensitiveDataFromExpiredDrafts();
+        
+        const expiredReports = await ComplianceReport.find({ 
+          expirationDate: { $lt: new Date() } 
+        }).exec();
+      
+        const nonExpiredReports = await ComplianceReport.find({ 
+          expirationDate: { $gte: new Date() } 
+        }).exec();
+        
+        const reportsAfterUpdate = await ComplianceReport.find({}).exec();
+
+        expiredReports.forEach(report => {
+            const reportObj = report.toObject();
+          
+            expect(reportObj).to.have.property('link');
+            expect(reportObj).to.have.property('expirationDate');
+          
+            const sensitiveFields = ["softwareName", "logo", "website", "documentation", "pointOfContact", "compliance", "status"];
+            sensitiveFields.forEach(field => {
+              expect(reportObj[field]).to.satisfy(val => val === undefined || val === null || (Array.isArray(val) && val.length === 0));
+            });
+          });
+      
+          nonExpiredReports.forEach(report => {
+            const expectedRequiredKeys = ['_id', 'softwareName', 'logo', 'website', 'documentation', 'pointOfContact', 'status', 'compliance'];
+            const expectedOptionalKeys = ['__v', 'link', 'expirationDate'];
+            const allExpectedKeys = [...expectedRequiredKeys, ...expectedOptionalKeys];
+          
+            const reportObj = report.toObject();
+            const reportKeys = Object.keys(reportObj);
+          
+            expectedRequiredKeys.forEach(key => {
+              expect(reportObj).to.have.property(key);
+            });
+          
+            reportKeys.forEach(key => {
+              expect(allExpectedKeys).to.include(key);
+            });
+          });
+    });
```

Comment:
checks inside these expired and nonExpired are quite similar, you could extract protected function e.g. "reportHasKeys(collection, keySet)"  and use it in both.

Reply:
Okay I will move it to testUtils file and create private function.